src/snowflake/snowpark/stored_procedure_profiler.py

```diff
@@ -69,18 +69,16 @@ def set_active_profiler(
 
         Args:
             active_profiler_type: String that represent active_profiler, must be either 'LINE' or 'MEMORY'
-            (case-sensitive). Active profiler is 'LINE' by default.
+            (case-insensitive). Active profiler is 'LINE' by default.
 
         """
         with self._lock:
             self._active_profiler_number += 1
-        if active_profiler_type not in ["LINE", "MEMORY"]:
+        if active_profiler_type.upper() not in ["LINE", "MEMORY"]:
             raise ValueError(
                 f"active_profiler expect 'LINE', 'MEMORY', got {active_profiler_type} instead"
             )
-        sql_statement = (
-            f"alter session set ACTIVE_PYTHON_PROFILER = '{active_profiler_type}'"
-        )
+        sql_statement = f"alter session set ACTIVE_PYTHON_PROFILER = '{active_profiler_type.upper()}'"
         self._session.sql(sql_statement)._internal_collect_with_tag_no_telemetry()
         with self._lock:
             if self._query_history is None:
```

Comment:
Does this assign a reference to _query_history?  If sesson.query_history changes, this variable will pick up the changes?